pennylane/qnode.py

```diff
@@ -1024,35 +1036,15 @@ def _pd_analytic(self, params, idx, force_order2=False, **kwargs):
                 B = np.eye(1 +2*w)
                 B_inv = B.copy()
                 for BB in self._op_successors(o_idx, 'G'):
-                    temp = BB.heisenberg_tr(w)
-                    B = temp @ B
-                    temp = BB.heisenberg_tr(w, inverse=True)
-                    B_inv = B_inv @ temp
+                    B = BB.heisenberg_tr(w) @ B
+                    B_inv = B_inv @ BB.heisenberg_tr(w, inverse=True)
                 Z = B @ Z @ B_inv  # conjugation
 
-                # ev_successors = self._op_successors(o_idx, 'E')
-
-                def tr_obs(ex):
-                    """Transform the observable"""
-                    # TODO: At initial release, since we use a queue to represent circuit, all expectations values
-                    # are successors to all gates in the same circuit.
-                    # When library uses a DAG representation for circuits, uncomment following if statement
-
-                    ## if ex is not a successor of op, multiplying by Z should do nothing.
-                    #if ex not in ev_successors:
-                    #    return ex
```

Comment:
@co9olguy, @smite: uncommenting the above breaks existing Gaussian jacobian tests. In some cases (not all), observables that should be transformed are not being transformed. @zzzeid can, you let us know the name of the test(s) that fail? One I believe is `test_template_layers.TestInterferometer.test_integration()`.

Before we merge, we should work out what is happening here. From my understanding, this comment is correct; an observable that is not a successor does not need to be transformed.

Reply:
By not uncommenting this `if` statement, we lose one of the major benefits of using a DAG